### DISCUSSION TITLE
Type-safety for returned type of not-null filter on relations

### DISCUSSION THREAD

**Discussion by DamianGlowala** [prisma/prisma#16814] [Q&A]
Given the following code (with `"strickNullChecks": true`):
```
const sampleUser = await this.prisma.userAccount.findFirst({
    where: {
        userRelatedTableName: {
            isNot: null
        }
    },
    select: {
        userRelatedTableName: true
    }
});
```
... the return type is:
```
const sampleUser: {
    userRelatedTableName: UserRelatedTableName | null;
} | null
```

Is it feasible for Prisma to achieve the following, not-null type based on the `where: { (...) }` clause?
```
const sampleUser: {
    userRelatedTableName: UserRelatedTableName;
} | null
```

**Comment by pantharshit00** [comment]
Hello @DamianGlowala,

Sorry for very late reply here. Looks like this issue was lost and it was recently converted into a discussion. 


But I think you should open a [feature request ](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) about this since currently the types which client ships doesn't account for this case. It might be the case that adding this feature can create an exponential increase in the generated types so that will be discussed in the feature request by someone working on that part of Prisma.

**Reply by LonJonn** [reply]
I've opened a draft PR related to this issue [here](https://github.com/prisma/prisma/pull/16883). It's only a hacky PoC but hopefully it can act as a base.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
